DatadogCore/Tests/Datadog/LoggerTests.swift

```diff
@@ -234,6 +235,38 @@ class LoggerTests: XCTestCase {
             logMatcher.assertValue(forKeyPath: "error.kind", equals: errorKind)
             logMatcher.assertValue(forKeyPath: "error.message", equals: errorMessage)
             logMatcher.assertValue(forKeyPath: "error.stack", equals: stackTrace)
+            logMatcher.assertValue(forKeyPath: "error.source_type", equals: "ios")
+        }
+    }
+
+    func testLoggingErrorWithSourceType() throws {
+        core.context = .mockAny()
+
+        let feature: LogsFeature = .mockAny()
+        try core.register(feature: feature)
+
+        let logger = Logger.create(in: core)
+        let errorKind = String.mockRandom()
+        let errorMessage = String.mockRandom()
+        let stackTrace = String.mockRandom()
+        logger._internal.log(level: .info,
+                   message: .mockAny(),
+                   errorKind: errorKind,
+                   errorMessage: errorMessage,
+                   stackTrace: stackTrace,
+                   attributes: [
+                    "_dd.error.source_type": "flutter"
+                   ]
```

Comment:
super nit: formatting :) (not a blocker)

line should be broken after `log(`

Reply:
🤔 Why didn't swiftlint complain?